### PR TITLE
Maintenance cleanups from feature/vtpm branch

### PR DESCRIPTION
**Pull request by psafont**
While feature/vtpm branch was intended as a  development branch for breaking changes for VTPM behaviour, this never materialized.

Some commits however ended up in the branch, all of them are maintenance cleanups and have been reviewed.

These commits have been cherry-picked because there were merges in the feature/vtpm branch, but this allows to squash the fixup commits.

Obsoletes #5032

### CODE REVIEW

File: ocaml/quicktest/quicktest_fe_systemctl.ml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      Filename.concat home ".config/systemd/user"
+  in
+  Filename.concat dir "xapi-envtest@.service"
+
+let get_temp_file prefix suffix =
+  (* /tmp visible to systemd may not match /tmp visible to test
+     when running tests inside containers.
+     Use a path in the current dir instead for testing.
+  *)
+  let name = Filename.temp_file ~temp_dir:(Sys.getcwd ()) prefix suffix in
```
Comment (lindig) [inline ocaml/quicktest/quicktest_fe_systemctl.ml:31]: `Filename.temp_file ` by default creates a name inside the temp file directory of the system. You are providing a different default here. Would in this case it not be enough to provide "." for the current directory?

**Review by lindig — APPROVED** [review]

**Comment by edwintorok** [comment]
Can you drop [[maintenance] fe_test.sh: reduce minimum sleep time](https://github.com/xapi-project/xen-api/pull/5033/commits/7e522386eb8afbd707c8f002bc3d3cb45a245750)? That is part of the maintenance PR and fixes some other issues with fe_test too.

**Comment by edwintorok** [comment]
In fact most of these commits are part of my maintenance PR except one fixup, I'll add that to my maintenance PR.

**Comment by psafont** [comment]
I'll close this PR, then

**Pull request closed without merging**